src/ch05-02-example-structs.md

```diff
@@ -1,266 +1,286 @@
 
-<span class="filename">Filename: src/main.rs</span>
+<span class="filename">Plik: src/main.rs</span>
 
 ```rust
-struct Rectangle {
-    width: u32,
-    height: u32,
+struct Prostokat {
+    szerokosc: u32,
+    wysokosc: u32,
 }
 
 fn main() {
-    let rect1 = Rectangle { width: 30, height: 50 };
+    let rect1 = Prostokat { szerokosc: 30, wysokosc: 50 };
 
     println!(
-        "The area of the rectangle is {} square pixels.",
-        area(&rect1)
+        "Pole prostokąta wynosi {} pikseli kwadratowych."
+        pole(&rect1)
     );
 }
 
-fn area(rectangle: &Rectangle) -> u32 {
-    rectangle.width * rectangle.height
+fn pole(prostokat: &Prostokat) -> u32 {
+    prostokat.width * prostokat.height
 }
 ```
 
-<span class="caption">Listing 5-10: Defining a `Rectangle` struct</span>
-
-Here we’ve defined a struct and named it `Rectangle`. Inside the curly
-brackets, we defined the fields as `width` and `height`, both of which have
-type `u32`. Then in `main`, we created a particular instance of `Rectangle`
-that has a width of 30 and a height of 50.
-
-Our `area` function is now defined with one parameter, which we’ve named
-`rectangle`, whose type is an immutable borrow of a struct `Rectangle`
-instance. As mentioned in Chapter 4, we want to borrow the struct rather than
-take ownership of it. This way, `main` retains its ownership and can continue
-using `rect1`, which is the reason we use the `&` in the function signature and
-where we call the function.
-
-The `area` function accesses the `width` and `height` fields of the `Rectangle`
-instance. Our function signature for `area` now says exactly what we mean:
-calculate the area of `Rectangle`, using its `width` and `height` fields. This
-conveys that the width and height are related to each other, and it gives
-descriptive names to the values rather than using the tuple index values of `0`
-and `1`. This is a win for clarity.
-
-### Adding Useful Functionality with Derived Traits
-
-It’d be nice to be able to print an instance of `Rectangle` while we’re
-debugging our program and see the values for all its fields. Listing 5-11 tries
-using the `println!` macro as we have used in previous chapters. This won’t
-work, however.
-
-<span class="filename">Filename: src/main.rs</span>
+<span class="caption">Listing 5-10: Definicja struktury `Prostokat`.
+
+Powyżej zdefiniowaliśmy strukturę i nazwaliśmy ją `Prostokat`.
+Wewnątrz nawiasów klamrowych zdefiniowaliśmy atrybuty `szerokosc` i `wysokosc`,
+oba mające typ `u32`.
+Następnie w funkcji `main` stworzyliśmy konkretną instancję struktury `Prostokat`,
+gdzie szerokość wynosi 30 jednostek i wysokość 50 jednostek.
+
+Nasza funkcja `area` przyjmuje teraz jeden parametr,
+który nazwaliśmy `prostokat`, którego typ to niezmienne zapożyczenie
+instancji struktury `Prostokat`.
+Jak wspomnieliśmy w Rozdziale 4, chcemy jedynie pożyczyć strukturę zamiast
+wejść w jej posiadanie. Takim sposobem `main` pozostaje właścicielem i może dalej
```

Comment:
"chcemy jedynie pożyczyć strukturę zamiast wejść w jej posiadanie" - wiem o co chodzi, ale jakoś glupio to brzmi.

Reply:
no tu się zgadzam, ale jak wspomniałaś czasem techniczny angielski ciężko przetłumaczyć, kilka pomysłów na szybko:

1. `Jak wspomnieliśmy w Rozdziale 4, chcemy uzyskać referencję do struktury zamiast zostania jej właścicielem`.
2. `..., wystarczy nam referencja do struktury, niekoniecznie chcemy wejść w jej posiadanie`

co sądzisz?

Reply:
A mnie się wydaje że "pożyczanie struktury" jest ok.
Oczywiście że brzmi to dla nas dziwnie, bo w żadnym innym języku programowania w ten sposób się nie mówiło. Ale w Rust'cie używa się takiego języka i dla nieprzyzwyczajonych tak samo dziwnie brzmi to po polsku jak i po angielsku.

Reply:
Ogólnie nie będę się upierać, ale bardziej chodziło mi o zmianę "zamiast wejść w jej posiadanie" na coś w stylu "bez przenoszenia własności". Zostawiam decyzję autorowi

Reply:
Brzmi lepiej, zmienię!